components/user_settings/notifications/email_notification_setting/email_notification_setting.tsx

```diff
@@ -14,27 +13,37 @@ import SettingItemMin from 'components/setting_item_min';
 
 const SECONDS_PER_MINUTE = 60;
 
-export default class EmailNotificationSetting extends React.PureComponent {
-    static propTypes = {
-        currentUserId: PropTypes.string.isRequired,
-        activeSection: PropTypes.string.isRequired,
-        updateSection: PropTypes.func.isRequired,
-        enableEmail: PropTypes.bool.isRequired,
-        emailInterval: PropTypes.number.isRequired,
-        onSubmit: PropTypes.func.isRequired,
-        onCancel: PropTypes.func.isRequired,
-        onChange: PropTypes.func.isRequired,
-        serverError: PropTypes.string,
-        saving: PropTypes.bool,
-        focused: PropTypes.bool,
-        sendEmailNotifications: PropTypes.bool,
-        enableEmailBatching: PropTypes.bool,
-        actions: PropTypes.shape({
-            savePreferences: PropTypes.func.isRequired,
-        }).isRequired,
+type Props = {
+    currentUserId: string;
+    activeSection: string;
+    updateSection: (section: string) => void;
+    enableEmail: boolean;
+    emailInterval: number;
+    onSubmit: () => void;
+    onCancel: () => void;
+    onChange: (enableEmail: boolean) => void;
+    serverError?: string;
+    saving?: boolean;
+    focused?: boolean;
+    sendEmailNotifications: boolean;
+    enableEmailBatching: boolean;
+    actions: {
+        savePreferences: (currentUserId: string, emailIntervalPreference: {}) =>
+        Promise<{}>;
```

Comment:
Maybe is too verbose here, since we do not use the value, but if we wanted to be strict, the promise result is also not {}, but {data: bool} (if I am not mistaken).